internal/cloud/realm/app.go

```diff
@@ -0,0 +1,137 @@
+package realm
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/10gen/realm-cli/internal/utils/api"
+)
+
+// UserProfile contains all of the roles associated with a user
+type UserProfile struct {
+	Roles []Role `json:"roles"`
+}
+
+// Role contains a GrouID field that maps to a project ID
+type Role struct {
+	GroupID string `json:"group_id"`
+}
+
+// App represents basic Realm App data
+type App struct {
+	ID          string `json:"_id"`
+	GroupID     string `json:"group_id"`
+	ClientAppID string `json:"client_app_id"`
+	Name        string `json:"name"`
+}
+
+var (
+	errGroupNotFound = errors.New("group could not be found")
+)
+
+var (
+	appsByGroupIDEndpoint = adminAPI + "/groups/%s/apps"
+	userProfileEndpoint   = adminAPI + "/auth/profile"
+)
+
+// FilterGroupIDsFromUserProfile returns all available group ids for a given user
+func (pd *UserProfile) FilterGroupIDsFromUserProfile(projectFlag string) []string {
+	groupIDs := []string{}
+	set := make(map[string]bool)
+	for _, role := range pd.Roles {
+		// Break early if we find the groupID that matches the project flag
+		if len(projectFlag) == 0 && !set[role.GroupID] {
+			groupIDs = append(groupIDs, role.GroupID)
+			set[role.GroupID] = true
+			continue
+		}
+
+		if role.GroupID == projectFlag {
+			return []string{role.GroupID}
+		}
+	}
+
+	return groupIDs
+}
+
+func (c *client) GetUserProfile() (UserProfile, error) {
```

Comment:
let's move `GetUserProfile` to the `auth.go` file of this package (along with its `struct`s, it's probably worth including the `AllGroupIDs() []string` function on the `UserProfile` type that the old `realm-cli` had)

Reply:
Done!